docs(mesh): in Vault docs switch to orphan tokens (#6964)
Kong/docs.konghq.com: app/_src/mesh/features/vault.md

#### Step 4. Configure authentication method:

> To authorize {{site.mesh_product_name}} to vault using a token, generate the following orphan token and pass it to mesh:

Reviewer: ```suggestion
To authorize {{site.mesh_product_name}} to vault using a token, generate the following orphan token and pass it to the mesh:
```

```sh
vault token create -type=service -orphan -format=json -policy="kmesh-default-dataplane-proxies" | jq -r ".auth.client_token"
```

> We suggest using an [orphan token](https://developer.hashicorp.com/vault/docs/concepts/tokens#token-hierarchies-and-orphan-tokens)
> to avoid surprising behavior around expiration. Creating them requires root/sudo
> permissions, Therefore you need those permissions to execute the above command. If you understand the
> implications, you can use a non-orphan token as well.

Reviewer: ```suggestion
We suggest using an [orphan token](https://developer.hashicorp.com/vault/docs/concepts/tokens#token-hierarchies-and-orphan-tokens)
to avoid surprising behavior around expiration. You need root/sudo
permissions to execute the previous command. If you understand the
implications, you can use a non-orphan token as well.
```
@michaelbeaumont What are the implications?

Author: Really just the surprising behavior around expiration, it's described a little more in the link

Author: @cloudjumpercat I simplified the language a little bit more
